### PR TITLE
fix: collect push bytes from test setup in fuzzer

### PR DESCRIPTION
**Pull request by onbjerg**
## Motivation

There's an old report that immutables were not collected in the fuzzer. Immutables are sections in the bytecode of contracts that are replaced by constant values, so we *should* be collecting them (since we collect push bytes), but it turns out that we were not collecting push bytes in the initial fuzzer DB (i.e. from the state right after `setUp`).

See https://github.com/foundry-rs/foundry/issues/1168

## Solution

Also collect push bytes after `setUp`. Closes https://github.com/foundry-rs/foundry/issues/1168

Note: The state dict after `setUp` for the test in #1168 contains about ~120 values. It seems that either one of the tests fail every time (which is what we want), but not super reliably. Increasing the fuzzer runs to 512 makes them both fail reliably, though.

### CODE REVIEW

File: evm/src/fuzz/strategies/state.rs [hunk 1]
```diff
@@ -87,6 +87,17 @@ pub fn build_initial_state<DB: DatabaseRef>(db: &CacheDB<DB>) -> EvmFuzzState {
         // Insert basic account information
         state.insert(H256::from(*address).into());
 
+        // Insert push bytes
+        if let Some(code) = &account.info.code {
+            if !state.cache.contains(address) {
+                state.cache.insert(*address);
```
Comment (mattsse) [inline evm/src/fuzz/strategies/state.rs:93]: can do this in one step

```suggestion
            if state.cache.insert(*address) {
```

**Review by mattsse — APPROVED** [review]
smol nit

**Comment by mattsse** [comment]
unsure if this is related but this now results in

```console
Failing tests:
Encountered 2 failing tests in src/test/StdAssertions.t.sol:StdAssertionsTest
[FAIL. Reason: Too many global rejects] testAssertEq_BytesErr_Pass(bytes,bytes) (runs: 31, μ: 20023, ~: 21439)
[FAIL. Reason: Too many global rejects] testAssertEq_Bytes_Pass(bytes,bytes) (runs: 24, μ: 19965, ~: 20697)
```

when running `forge test` on `forge-std`, I could reproduce this locally

**Comment by joshieDo** [comment]
We had this discussion here: https://github.com/foundry-rs/foundry/pull/2724#issuecomment-1212440233

We decided to revert the `push` collection on `build_initial_state`. But I'm all for keeping this and change the `forge-std` test instead.

**Comment by onbjerg** [comment]
The Forge Std `assume` call is just way too narrow so I'd expect it to reach the limit, and the fact that the reject limit hasn't been reached before is not ideal. Either they can increase the reject limit in Forge Std or just fix the test, obviously if the `assume` call makes it so `a` is always `b` then they can just take a single parameter.

**Comment by mds1** [comment]
Just merged https://github.com/foundry-rs/forge-std/pull/160 which should prevent foundry CI from failing on those now

**Review by gakonst — APPROVED** [review]
Rebased, merging per the forge-std fixes. Flagging for us to keep this PR in mind if people come to us with complaints about the fuzzer error'ing too much.

**Pull request merged**